src/rulesets/Base/sort.jl

```diff
@@ -0,0 +1,40 @@
+function rrule(::typeof(partialsort), xs::AbstractVector, k::Union{Integer,OrdinalRange}; kwargs...)
+    inds = partialsortperm(xs, k; kwargs...)
+    ys = xs[inds]
+
+    function partialsort_pullback(Δys)
+        function partialsort_add!(Δxs)
+            for (Δy, i) in zip(Δys, inds)
+                Δxs[i] += Δy
+            end
+            return Δxs
+        end
+
+        Δxs = InplaceableThunk(
+            @thunk(partialsort_add!(zero(xs))),
```

Comment:
```suggestion
            @thunk(partialsort_add!(zero(Δys))),
```

I think it would be better to do `Δys` than `xs` to get our zeros from.
that way we don't close over `xs`
also because `Δys` definately has the right type to hold these results.
Which could matter for things with dervivative that are not the same type as them.

Reply:
Done this for `sort`.

However, in `partialsort` `Δys` can be a float (if `k` is an integer), so I think we need `zero(xs)`?

Reply:
fair